backend/app/services/context/context_service.py

```
@@ -0,0 +1,695 @@
+
+        try:
+            # Save binary data to storage backend
+            metadata = {
+                "filename": filename,
+                "mime_type": mime_type,
+                "file_size": file_size,
+                "user_id": user_id,
+            }
+            storage_backend.save(storage_key, binary_data, metadata)
+        except StorageError as e:
+            logger.error(f"Failed to save context {context.id} to storage: {e}")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)